__tests__/search.js

```diff
@@ -35,19 +35,77 @@ class MockVFSServiceProvider extends ServiceProvider {
     ];
   }
 }
+class MySearchAdapter {
+    constructor(core) {
+        this.core = core;
 
+    }
+
+    destroy() {
+    }
+
+    async init() {
+    }
+
+    async search(pattern) {
+        const obj = [
+            {
+                filename: "mahdi.png",
+                humanSize: "5.9 KiB"+pattern,
+                icon: null,
+                id: null,
+                isDirectory: false,
+                isFile: true,
+                label: null,
+                mime: "image/png",
+                parent_id: null,
+                path: "hhh",
+                size: 1959,
+                stat: {
+                    atime: "2022-11-26T22:06:27.343Z",
+                    atimeMs: 1669500387343,
+                    birthtime: "2022-11-26T22:06:27.049Z",
+                    birthtimeMs: 1669500387049.1592,
+                    blksize: 4096,
+                    blocks: 8,
+                    ctime: "2022-11-26T22:06:27.389Z",
+                    ctimeMs: 1669500387389.1677,
+                    dev: 473163771,
+                    gid: 0,
+                    ino: 3377699720581588,
+                    mode: 3320,
+                    mtime: "1985-10-26T08:15:00.000Z",
+                    mtimeMs: 499162500000,
+                    nlink: 1,
+                    rdev: 0,
+                    size: 1959,
+                    uid: 0
+                }
+            }
+        ];
+        return Promise.all(obj)
+            .then(lists => [].concat(...lists));
+    }
+}
 describe('Search', () => {
   let core;
   let search;
-
+  let options = {};
+  options = {
+        args: {
```

Comment:
There is no `args` in the API. As I wrote in the previous review *you* made this `options = { adapters: [] }`